external/llvm-project/mlir/test/Dialect/AMDGPU/canonicalize.mlir

```diff
@@ -0,0 +1,132 @@
+// RUN: mlir-opt %s -split-input-file -canonicalize  | FileCheck %s
+
+// CHECK-LABEL: func @known_oob_load
+func.func @known_oob_load(%arg0: memref<4xf32>) -> f32 {
+  // CHECK: %[[zero:.*]] = arith.constant 0.000000e+00 : f32
+  // CHECK: return %[[zero]]
```

Comment:
Nit: I think this applies to cases below as well. It'd be much robust if you use `CHECK-NOT` instead to explicitly reject the raw_buffer_load.

Reply:
As in, CHECK-NOT in addition to the existing check or instead of it? 

I worded the test this way because I wanted to check the exact semantics of the rewrites

Reply:
Yeah, I can see what you did make sense. I think this is just a style thing that doesn't impact the correctness of testing. Feel free to leave it.

My reasoning is that since we get rid of a instruction, we should not test whether zero is correctly returned (which should be covered in arith dialect).

Reply:
My thoughts were that the function originally returns the value of the load, so I wanted to make sure that the return value got replaced by 0 like it should be.